src/main/scala/scala/tools/refactoring/sourcegen/ReusingPrinter.scala

```diff
@@ -553,14 +606,30 @@ trait ReusingPrinter extends TreePrintingTraversals with AbstractPrinter {
           }
 
         case (fun, Nil) =>
-
-          // Calls to methods without `()` are represented by a select and no apply.
-          if (r.matches("""^\s*\)""")) {
-            l ++ p(fun) ++ "(" ++ r
-          } else {
-            l ++ p(fun) ++ r
+          val trailingLayout: Layout = {
+            val rr = r
+            if (rr.nonEmpty) {
+              if ((commentsAndSpaces ~ ')').consumes(rr.asText)) {
+                // Needed for IndividualSourceGenTest.testPrintParentheses
+                Layout("(") ++ rr
+              } else {
+                rr
+              }
+            } else {
+              fun.pos match {
+                case pos: RangePosition =>
+                  // If fun is part of an expression involving default arguments, it might be
+                  // necessary to manually make sure that parenthesis and '.' is printed out.
+                  // See #1002564
+                  val parensAndSep = commentsAndSpaces ~ '(' ~ commentsAndSpaces ~ ')' ~ '.'.optional
+                  Layout(Movement.coveredStringStartingAtEndOf(pos, parensAndSep))
```

Comment:
Thanks :-).. I think that there is an error though, as the optional `'.'` might be proceeded by `commentsAndSpaces`, as `foo /*strange*/  . /*???*/  bar()` is legal Scala. In fact, I'd like to go for something like `('(' ~ ')' ~ '.'.optional).skipping(commentsAndSpaces)`... it might make sense to do this in a separate PR though.